lib/routes/authorization.js

```diff
@@ -29,15 +35,30 @@ function set(arr) {
   return Object.keys(obj);
 }
 
-
+function scopeStringToSet(scopes) {
+  return typeof scopes === 'string' ?
+    set(scopes.split(' ')) :
+    scopes;
+}
 
 function isLocalHost(url) {
   var host = new URI(url).hostname();
   return host === 'localhost' || host === '127.0.0.1';
 }
 
-function generateCode(claims, client, scope, req) {
+function scopeCheck(requestedScopes, validScopes) {
```

Comment:
nit: I suggest naming this something a little more specific, since to me "scope check" means "check whether a particular scope is included in this list of scopes" rather than "check these scopes for validity".